### ISSUE TITLE
[java] requirement to print matched line instead of/along with defined static message attribute in rule xml

### ISSUE THREAD

**Issue by rajeshveera** [pmd/pmd#2346]
Using Version:6.17.0
 
####Use case####
Need to identify exact matched element of List[String] configured in rule xml.Exact violation require to be pinpointed in pmd report itself as each element of list[String] will have multiple owners to track.Generic message will not help them.

As mentioned below , any methodname of list of methodnames is matched pmd report shown message along with line.But seems that there is no option to check 'which methodname is matched' in line.PMD enforce to write custom pmd rule class just to identify matched content.PMD report is already showing matched line , so it is possible for PMD to print matched line too.

<rule name="MyXpathRule" language="java" class="net.sourceforge.pmd.lang.rule.XPathRule" >
<description>
 Core table usage restriction
 </description>
 <priority>1</priority>
  <properties>
    <property name="version" value="2.0" />
    <property name="methodnames" type="List[String]" value="method1,method2,method3" delimiter=","  description="method usage restriction" />
    <property name="xpath">
    <value>
    <![CDATA[
   //Arguments//PrimaryPrefix/Name[@Image=$methodnames]
    ]]>
    </value>
  </property>
  </properties>
</rule>

#####Suggestion####
Say  if new 'matchedline' is true then matched line can be printed along with default message attribute value.It could prevent PMD user from writing new custom rule class .

**Comment by jsotuyod** [comment]
@rajeshveera what report format are you using? Could you provide an example to make this more evident?

If I'm understanding correctly, for instance the XML report should be fine… violations are reported such as:

For the code:
```java
public class MyActivity {

    public MyActivity() {}
}
```

The report is:
```xml
<violation beginline="3" endline="3" begincolumn="12" endcolumn="27" rule="UncommentedEmptyConstructor" ruleset="Documentation" package="com.foo.bar" class="MyActivity" method="MyActivity" externalInfoUrl="https://pmd.github.io/pmd-6.5.0/pmd_rules_java_documentation.html#uncommentedemptyconstructor" priority="3">
Document empty constructor
</violation
```

stating not only the message, but also:
* being line
* end line
* begin column
* end column

These properties correspond to the matched element. As you mention `List[String]`, make sure that the selected node is the proper one, and you are not selecting a parent node and just by filtering it's children's contents.
